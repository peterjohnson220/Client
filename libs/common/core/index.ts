export * from './pipes';
export * from './helpers';
