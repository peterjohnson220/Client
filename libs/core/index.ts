export * from './helpers';
export * from './pipes';
export * from './common.module';
