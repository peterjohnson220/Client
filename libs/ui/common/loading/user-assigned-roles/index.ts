export * from './user-assigned-roles.component';
