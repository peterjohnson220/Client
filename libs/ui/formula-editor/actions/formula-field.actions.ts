import { Action } from '@ngrx/store';

import { DataViewField } from 'libs/models/payfactors-api';
import { FormulaFieldModalObj } from 'libs/models/formula-editor';

import { Field, FieldDataType, FormulaType } from '../models';

export const WAIT_FOR_FORMULA_VALIDATION = '[Data Insights / Formula Field Modal] Wait For Formula Validation';
export const VALIDATE_FORMULA = '[Data Insights / Formula Field Modal] Validate Formula';
export const VALIDATE_FORMULA_SUCCESS = '[Data Insights / Formula Field Modal] Validate Formula Success';
export const VALIDATE_FORMULA_ERROR = '[Data Insights / Formula Field Modal] Validate Formula Error';
export const SAVE_FORMULA_FIELD = '[Data Insights / Formula Field Modal] Save Formula Field';
export const CREATE_FORMULA_FIELD_SUCCESS = '[Data Insights / Formula Field Modal] Create Formula Field Success';
export const UPDATE_FORMULA_FIELD_SUCCESS = '[Data Insights / Formula Field Modal] Update Formula Field Success';
export const SAVE_FORMULA_FIELD_ERROR = '[Data Insights / Formula Field Modal] Save Formula Field Error';
export const RESET_MODAL = '[Data Insights / Formula Field Modal] Reset Modal';
export const GET_FORMULA_FIELD_VIEW_COUNT = '[Data Insights / Formula Field Modal] Get Formula Field View Count';
export const GET_FORMULA_FIELD_VIEW_COUNT_SUCCESS = '[Data Insights / Formula Field Modal] Get Formula Field View Count Success';
export const GET_FORMULA_FIELD_VIEW_COUNT_ERROR = '[Data Insights / Formula Field Modal] Get Formula Field View Count Error';
export const DELETE_FORMULA_FIELD = '[Data Insights / Formula Field Modal] Delete Formula Field';
export const DELETE_FORMULA_FIELD_SUCCESS = '[Data Insights / Formula Field Modal] Delete Formula Field Success';
export const DELETE_FORMULA_FIELD_ERROR = '[Data Insights / Formula Field Modal] Delete Formula Field Error';
export const RESET_FORMULA = '[Data Insights / Formula Field Modal] Reset Formula';

export class WaitForFormulaValidation implements Action {
  readonly type = WAIT_FOR_FORMULA_VALIDATION;

  constructor(public payload: { formulaFieldId: string }) {}
}

export class ValidateFormula implements Action {
  readonly type = VALIDATE_FORMULA;

  constructor(public payload: { formula: string, baseEntityId: number, formulaFieldId: string }) {}
}

export class ValidateFormulaSuccess implements Action {
  readonly type = VALIDATE_FORMULA_SUCCESS;

  constructor(public payload: { result: boolean, dataType: FieldDataType, formulaFieldId: string }) {}
}

export class ValidateFormulaError implements Action {
  readonly type = VALIDATE_FORMULA_ERROR;

  constructor(public payload: { formulaFieldId: string }) {}
}

export class SaveFormulaField implements Action {
  readonly type = SAVE_FORMULA_FIELD;

  constructor(public payload: { formula: FormulaFieldModalObj, baseEntityId: number, formulaFieldId: string, formulaTypeId: FormulaType }) {}
}

export class CreateFormulaFieldSuccess implements Action {
  readonly type = CREATE_FORMULA_FIELD_SUCCESS;

  constructor(public payload: { dataViewField: DataViewField, formulaFieldId: string }) {}
}

export class UpdateFormulaFieldSuccess implements Action {
  readonly type = UPDATE_FORMULA_FIELD_SUCCESS;

  constructor(public payload:  { dataViewField: DataViewField, formulaFieldId: string }) {}
}

export class SaveFormulaFieldError implements Action {
  readonly type = SAVE_FORMULA_FIELD_ERROR;

  constructor(public payload: { message: string, formulaFieldId: string }) {}
}

export class ResetModal implements Action {
  readonly type = RESET_MODAL;

  constructor(public payload: { formulaFieldId: string }) {}
}

export class DeleteFormulaField implements Action {
  readonly type = DELETE_FORMULA_FIELD;

  constructor(public payload: Field) {}
}

export class DeleteFormulaFieldSuccess implements Action {
  readonly type = DELETE_FORMULA_FIELD_SUCCESS;

  constructor(public payload: Field) {}
}

export class DeleteFormulaFieldError implements Action {
  readonly type = DELETE_FORMULA_FIELD_ERROR;

  constructor() {}
}

export class ResetFormula implements Action {
  readonly type = RESET_FORMULA;

  constructor(public payload: { formulaFieldId: string }) {}
}

export class GetFormulaFieldViewCount implements Action {
  readonly type = GET_FORMULA_FIELD_VIEW_COUNT;

  constructor(public payload: number) {}
}

export class GetFormulaFieldViewCountSuccess implements Action {
  readonly type = GET_FORMULA_FIELD_VIEW_COUNT_SUCCESS;

  constructor(public payload: number) {}
}

export class GetFormulaFieldViewCountError implements Action {
  readonly type = GET_FORMULA_FIELD_VIEW_COUNT_ERROR;

  constructor() {}
}

export type Actions
  = WaitForFormulaValidation
  | ValidateFormula
  | ValidateFormulaSuccess
  | ValidateFormulaError
  | SaveFormulaField
  | CreateFormulaFieldSuccess
  | UpdateFormulaFieldSuccess
  | SaveFormulaFieldError
  | ResetModal
  | DeleteFormulaField
  | DeleteFormulaFieldSuccess
  | DeleteFormulaFieldError
  | ResetFormula
  | GetFormulaFieldViewCount
  | GetFormulaFieldViewCountSuccess
  | GetFormulaFieldViewCountError;
