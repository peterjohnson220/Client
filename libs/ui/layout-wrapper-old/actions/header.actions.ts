import { Action } from '@ngrx/store';

import { NavigationLink } from '../../../models';

export const GET_HEADER_DROPDOWN_NAVIGATION_LINKS = '[Layout Wrapper/Header] Get Header Dropdown Navigation Links';
export const GET_HEADER_DROPDOWN_NAVIGATION_LINKS_SUCCESS = '[Layout Wrapper/Header] Get Header Dropdown Navigation Links Success';
export const GET_HEADER_DROPDOWN_NAVIGATION_LINKS_ERROR = '[Layout Wrapper/Header] Get Header Dropdown Navigation Links Error';

export class GetHeaderDropdownNavigationLinks implements Action {
  readonly type = GET_HEADER_DROPDOWN_NAVIGATION_LINKS;

  constructor() {}
}

export class GetHeaderDropdownNavigationLinksSuccess implements Action {
  readonly type = GET_HEADER_DROPDOWN_NAVIGATION_LINKS_SUCCESS;

  constructor(public payload: NavigationLink[]) {}
}

export class GetHeaderDropdownNavigationLinksError implements Action {
  readonly type = GET_HEADER_DROPDOWN_NAVIGATION_LINKS_ERROR;

  constructor() {}
}

export type Actions =
  | GetHeaderDropdownNavigationLinks
  | GetHeaderDropdownNavigationLinksSuccess
  | GetHeaderDropdownNavigationLinksError;
