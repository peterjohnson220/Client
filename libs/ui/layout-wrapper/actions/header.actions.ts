import { Action } from '@ngrx/store';

import { NavigationLink, HomePageLink } from '../../../models';

export const GET_HEADER_DROPDOWN_NAVIGATION_LINKS = '[Layout Wrapper/Header] Get Header Dropdown Navigation Links';
export const GET_HEADER_DROPDOWN_NAVIGATION_LINKS_SUCCESS = '[Layout Wrapper/Header] Get Header Dropdown Navigation Links Success';
export const GET_HEADER_DROPDOWN_NAVIGATION_LINKS_ERROR = '[Layout Wrapper/Header] Get Header Dropdown Navigation Links Error';

export const GET_HEADER_USER_HOMEPAGE_LINK = '[Layout Wrapper/Header] Get Header User Homepage Link';
export const GET_HEADER_USER_HOMEPAGE_LINK_SUCCESS = '[Layout Wrapper/Header] Get Header User Homepage Link Success';
export const GET_HEADER_USER_HOMEPAGE_LINK_ERROR = '[Layout Wrapper/Header] Get Header User Homepage Link Error';

export class GetHeaderDropdownNavigationLinks implements Action {
  readonly type = GET_HEADER_DROPDOWN_NAVIGATION_LINKS;

  constructor() {}
}

export class GetHeaderDropdownNavigationLinksSuccess implements Action {
  readonly type = GET_HEADER_DROPDOWN_NAVIGATION_LINKS_SUCCESS;

  constructor(public payload: NavigationLink[]) {}
}

export class GetHeaderDropdownNavigationLinksError implements Action {
  readonly type = GET_HEADER_DROPDOWN_NAVIGATION_LINKS_ERROR;

  constructor() {}
}

export class GetHeaderUserHomePageLink implements Action {
  readonly type = GET_HEADER_USER_HOMEPAGE_LINK;

  constructor(public payload: any) {}
}

export class GetHeaderUserHomePageLinkSuccess implements Action {
  readonly type = GET_HEADER_USER_HOMEPAGE_LINK_SUCCESS;

  constructor(public payload: HomePageLink) {}
}

export class GetHeaderUserHomePageLinkError implements Action {
  readonly type = GET_HEADER_USER_HOMEPAGE_LINK_ERROR;

  constructor() {}
}

export type Actions =
  | GetHeaderDropdownNavigationLinks
  | GetHeaderDropdownNavigationLinksSuccess
  | GetHeaderDropdownNavigationLinksError
  | GetHeaderUserHomePageLink
  | GetHeaderUserHomePageLinkSuccess
  | GetHeaderUserHomePageLinkError;
