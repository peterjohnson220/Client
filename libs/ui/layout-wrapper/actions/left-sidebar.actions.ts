import { Action } from '@ngrx/store';

import { SidebarLink } from '../../../models';

export const GET_LEFT_SIDEBAR_NAVIGATION_LINKS = '[Layout Wrapper/Left Sidebar] Get Left Sidebar Navigation Links';
export const GET_LEFT_SIDEBAR_NAVIGATION_LINKS_SUCCESS = '[Layout Wrapper/Left Sidebar] Get Left Sidebar Navigation Links Success';
export const GET_LEFT_SIDEBAR_NAVIGATION_LINKS_ERROR = '[Layout Wrapper/Left Sidebar] Get Left Sidebar Navigation Links Error';
export const TOGGLE_LEFT_SIDEBAR = '[Layout Wrapper/Left Sidebar] Toggle Left Sidebar';
export const CLOSE_LEFT_SIDEBAR = '[Layout Wrapper/Left Sidebar] Close Left Sidebar';
export const URL_REDIRECT_APPLICATION_SUCCESS = '[Layout Wrapper/Left Sidebar] Url Redirect Application Success';

export class GetLeftSidebarNavigationLinks implements Action {
  readonly type = GET_LEFT_SIDEBAR_NAVIGATION_LINKS;

  constructor() {}
}

export class GetLeftSidebarNavigationLinksSuccess implements Action {
  readonly type = GET_LEFT_SIDEBAR_NAVIGATION_LINKS_SUCCESS;

  constructor(public payload: SidebarLink[]) {}
}

export class GetLeftSidebarNavigationLinksError implements Action {
  readonly type = GET_LEFT_SIDEBAR_NAVIGATION_LINKS_ERROR;

  constructor() {}
}

export class ToggleLeftSidebar implements Action {
  readonly type = TOGGLE_LEFT_SIDEBAR;
  constructor(public payload: boolean) {}
}

export class CloseLeftSidebar implements Action {
  readonly type = CLOSE_LEFT_SIDEBAR;
  constructor() {}
}

export class UrlRedirectApplicationSuccess implements Action {
  readonly type = URL_REDIRECT_APPLICATION_SUCCESS;
  constructor(public payload: SidebarLink[]) {
  }
}

export type Actions =
  | GetLeftSidebarNavigationLinks
  | GetLeftSidebarNavigationLinksSuccess
  | GetLeftSidebarNavigationLinksError
  | ToggleLeftSidebar
  | CloseLeftSidebar
  | UrlRedirectApplicationSuccess;
