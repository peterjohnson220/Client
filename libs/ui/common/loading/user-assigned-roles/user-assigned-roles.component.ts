import { Component, OnInit } from '@angular/core';

import { Store } from '@ngrx/store';
import { Observable } from 'rxjs';

import * as fromUserAssignedRoleActions from 'libs/state/app-context/actions/user-assigned-roles.actions';
import * as fromRootState from 'libs/state/state';

import {UserAssignedRole} from '../../../../models';

@Component({
  selector: 'pf-loading-user-assigned-roles',
  templateUrl: './user-assigned-roles.component.html'
})

export class UserAssignedRolesComponent implements OnInit {
  gettingUserAssignedRoles$: Observable<boolean>;
  userAssignedRoles$: Observable<UserAssignedRole[]>;

  constructor(
    private store: Store<fromRootState.State>
  ) {
    this.gettingUserAssignedRoles$ = store.select(fromRootState.getUserAssignedRolesLoading);
    this.userAssignedRoles$ = store.select(fromRootState.getUserAssignedRoles);
  }
  ngOnInit() {
    this.store.dispatch(new fromUserAssignedRoleActions.GetUserAssignedRoles());
  }
}

