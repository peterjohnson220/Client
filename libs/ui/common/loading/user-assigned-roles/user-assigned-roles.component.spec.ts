import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';

import { StoreModule, Store } from '@ngrx/store';
import { of } from 'rxjs/observable/of';

import * as fromRootState from 'libs/state/state';
import * as fromUserAssignedRoleActions from 'libs/state/app-context/actions/user-assigned-roles.actions';

import { UserAssignedRolesComponent} from './user-assigned-roles.component';


describe('Loading User Context', () => {
  let fixture: ComponentFixture<UserAssignedRolesComponent>;
  let instance: UserAssignedRolesComponent;
  let store: Store<fromRootState.State>;

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers
        }),
      ],
      declarations: [
        UserAssignedRolesComponent
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    store = TestBed.get(Store);

    spyOn(store, 'dispatch');

    fixture = TestBed.createComponent(UserAssignedRolesComponent);
    instance = fixture.componentInstance;
  });

  it('should dispatch a GetUserAssignedRoles action upon Init', () => {
    const action = new fromUserAssignedRoleActions.GetUserAssignedRoles();

    fixture.detectChanges();

    expect(store.dispatch).toHaveBeenCalledWith(action);
  });

  it('should show a centered loading indicator, when we are getting the user assigned roles', () => {
    instance.gettingUserAssignedRoles$ = of(true);

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should show a router outlet, when we are not getting the user assigned roles', () => {
    instance.gettingUserAssignedRoles$ = of(false);

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

});
