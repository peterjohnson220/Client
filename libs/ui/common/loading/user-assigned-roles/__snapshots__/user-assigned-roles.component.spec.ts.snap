// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading User Context should show a centered loading indicator, when we are getting the user assigned roles 1`] = `
<pf-loading-user-assigned-roles
  gettingUserAssignedRoles$={[Function Observable]}
  store={[Function Store]}
  userAssignedRoles$={[Function Store]}
>
  <div
    class="h-100 d-flex justify-content-center align-items-center"
  >
    <h2
      class="text-white"
    >
      <pf-loading-indicator />
    </h2>
  </div>
</pf-loading-user-assigned-roles>
`;

exports[`Loading User Context should show a router outlet, when we are not getting the user assigned roles 1`] = `
<pf-loading-user-assigned-roles
  gettingUserAssignedRoles$={[Function Observable]}
  store={[Function Store]}
  userAssignedRoles$={[Function Store]}
>
  <router-outlet />
</pf-loading-user-assigned-roles>
`;
